### PR TITLE
Mbarba/cleanup

### PR DESCRIPTION
**Pull request by MatBarba**
Review of code from mypy/pylint outputs (mostly formatting and type hints fixing)

### CODE REVIEW

File: src/python/ensembl/io/genomio/seq_region_dumper.py [hunk 1]
```diff
@@ -203,7 +204,7 @@ class InputSchema(argschema.ArgSchema):
     brc_mode = argschema.fields.Bool(default=False, metadata={"description": "BRC specific output"})
 
 
-def make_mysql_url(host: str, user: str, database: str, port: str = 0, password: str = "") -> str:
+def make_mysql_url(host: str, user: str, database: str, port: int = 0, password: str = "") -> str:
```
Comment (JAlvarezJarreta) [inline src/python/ensembl/io/genomio/seq_region_dumper.py:207]: Nice catch!

File: src/python/ensembl/io/genomio/process_gff3.py [hunk 2]
```diff
@@ -194,34 +192,34 @@ def merge_genes(self, tomerge: List) -> str:
...
         ignored_gene_types = self.ignored_gene_types
         transcript_types = self.transcript_types
         allowed_non_gene_types = self.non_gene_types
         skip_unrecognized = self.skip_unrecognized
         to_exclude = self.exclude_seq_regions
-        
-        functional_annotation = []
-        
+
+        functional_annotation: List = []
```
Comment (JAlvarezJarreta) [inline src/python/ensembl/io/genomio/process_gff3.py:211]: Same as before.

File: src/python/ensembl/io/genomio/process_gff3.py [hunk 3]
```diff
@@ -194,34 +192,34 @@ def merge_genes(self, tomerge: List) -> str:
...
+
         with out_gff_path.open("w") as gff3_out:
             new_records = []
-            fail_types = {}
-            
+            fail_types: Dict = {}
```
Comment (JAlvarezJarreta) [inline src/python/ensembl/io/genomio/process_gff3.py:215]: Same as before.

File: src/python/ensembl/io/genomio/process_gff3.py [hunk 4]
```diff
@@ -291,17 +288,15 @@ def format_mobile_element(self, feat, functional_annotation):
...
         self.add_funcann_feature(functional_annotation, feat, "transposable_element")
-        feat.qualifiers = {
-            "ID": feat.id
-        }
+        feat.qualifiers = {"ID": feat.id}
 
         return feat
 
-    def normalize_gene(self, gene: SeqFeature, functional_annotation: List, fail_types: List) -> SeqFeature:
+    def normalize_gene(self, gene: SeqFeature, functional_annotation: List, fail_types: Dict) -> SeqFeature:
```
Comment (JAlvarezJarreta) [inline src/python/ensembl/io/genomio/process_gff3.py:299]: Same as before re. `List` and `Dict`.

File: src/python/ensembl/io/genomio/process_gff3.py [hunk 5]
```diff
@@ -547,30 +541,30 @@ def cds_gene(self, cds: SeqFeature) -> SeqFeature:
         exon = SeqFeature(cds.location, type="exon")
         exon.qualifiers["source"] = cds.qualifiers["source"]
         transcript.sub_features.append(exon)
-        
+
         # Create a gene, add the transcript
         gene_type = "gene"
-        if("pseudo" in cds.qualifiers and cds.qualifiers["pseudo"][0] == "true"):
+        if "pseudo" in cds.qualifiers and cds.qualifiers["pseudo"][0] == "true":
```
Comment (JAlvarezJarreta) [inline src/python/ensembl/io/genomio/process_gff3.py:547]:
```suggestion
        if ("pseudo" in cds.qualifiers) and (cds.qualifiers["pseudo"][0] == "true"):
```
This is just a suggestion that I find more readable (and safer).

File: src/python/ensembl/io/genomio/process_gff3.py [hunk 6]
```diff
@@ -940,18 +931,19 @@ def remove_prefixes(self, identifier: str, prefixes: List[str]) -> str:
...
     interim_gff_fh.seek(0)
     # Load gff3 data and write a simpler version that follows our specifications
     gff.simpler_gff3(interim_gff_fh, out_gff_path, out_funcann_path)
 
-if __name__=="__main__":
-    main()
\ No newline at end of file
+
+if __name__ == "__main__":
+    main()
```
Comment (JAlvarezJarreta) [inline src/python/ensembl/io/genomio/process_gff3.py:949]: Same as before.

**Review by JAlvarezJarreta — APPROVED** [review]
All my comments are more suggestions than actual things that need to be addressed (they may even be things we want to do, but they are outside the scope of this PR), so I will go ahead and approve :+1:

Hope you have enjoyed your first full `pylint` and `mypy` experience!

**Comment by MatBarba** [comment]
I'd rather we keep the main call, just so we can call the script directly in dev.
As for MySQLConnection, I'll keep it as Any because I don't want to try to spend time on this (also I will retire core_server since we can use ensembl-py for the database connection now).

File: src/python/ensembl/io/genomio/manifest_stats.py [hunk 7]
```diff
@@ -279,31 +271,50 @@ def compare_ncbi_counts(self, prepared: Dict, ncbi: Dict) -> List:
...
+    )
+    accession = argschema.fields.String(
+        metadata={"description": "Sequence accession ID to compare stats with NCBI"}
+    )
+    datasets_bin = argschema.fields.String(metadata={"description": "Datasets bin status"})
+
 
 def main():
-    #args = get_args()
+    # args = get_args()
```
Comment (JAlvarezJarreta) [inline src/python/ensembl/io/genomio/manifest_stats.py:311]:
```suggestion
```

File: src/python/ensembl/io/genomio/events_dumper.py [hunk 8]
```diff
@@ -25,10 +28,41 @@
...
+class Pair:
+    """Simple old_id - new_id pair representation"""
+
+    def __init__(self, old_id: Optional[str], new_id: Optional[str]) -> None:
+        """Create a pair with an old_id and a new_id if provided"""
+
+        if old_id is not None:
+            self.old_id = old_id
+        else:
+            self.old_id = ""
```
Comment (JAlvarezJarreta) [inline src/python/ensembl/io/genomio/events_dumper.py:44]: Just for future reference, although I this alternative is more a preference than anything else:
```suggestion
        self.old_id = old_id if old_id is not None else ""
```

File: src/python/ensembl/io/genomio/process_gff3.py [hunk 9]
```diff
@@ -22,36 +22,37 @@
 import json
 import argschema
 
-from os import PathLike
-from typing import List, TextIO
+from typing import Dict, List, IO
 
-#import eHive
+# import eHive
```
[the remark this answers is not in the document]
Comment (JAlvarezJarreta) [reply src/python/ensembl/io/genomio/process_gff3.py:28]: I believe this one can go.

**Review by JAlvarezJarreta — APPROVED** [review]
2 comment that can be removed, but that's about it. Good to go :+1:

**Pull request merged**